frontend/public/components/monitoring/dashboards/custom-time-range-modal.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useDispatch } from 'react-redux';
+
+import { DatePicker, TimePicker } from '@patternfly/react-core';
+
+import {
+  monitoringDashboardsSetEndTime,
+  monitoringDashboardsSetTimespan,
+} from '../../../actions/ui';
+import {
+  createModalLauncher,
+  ModalBody,
+  ModalComponentProps,
+  ModalSubmitFooter,
+  ModalTitle,
+} from '../../factory/modal';
+import { dateFormatter } from '../../utils/datetime';
+
+// Get YYYY-MM-DD date string for a date object
+const toISODate = (date: Date): string => date.toISOString().split('T')[0];
+
+const CustomTimeRangeModal = ({ cancel, close }: ModalComponentProps) => {
+  const { t } = useTranslation();
+
+  const dispatch = useDispatch();
+
+  // Default to a time range that covers all of today
+  const now = new Date();
+  const [fromDate, setFromDate] = React.useState(now);
+  const [fromTime, setFromTime] = React.useState('00:00');
+  const [toDate, setToDate] = React.useState(now);
+  const [toTime, setToTime] = React.useState('23:59');
+
+  const submit: React.FormEventHandler<HTMLFormElement> = (e) => {
+    e.preventDefault();
+    const from = Date.parse(`${toISODate(fromDate)} ${fromTime}`);
+    const to = Date.parse(`${toISODate(toDate)} ${toTime}`);
+    dispatch(monitoringDashboardsSetEndTime(to));
+    dispatch(monitoringDashboardsSetTimespan(to - from));
+    close();
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle>{t('public~Custom time range')}</ModalTitle>
+      <ModalBody>
+        <div className="row co-m-form-row">
+          <div className="col-sm-12">
+            <label>{t('public~From')}</label>
+          </div>
+          <div className="col-sm-4">
+            <DatePicker
+              onChange={(str, date) => setFromDate(date)}
+              value={dateFormatter.format(fromDate)}
+            />
+          </div>
+          <div className="col-sm-4">
+            <TimePicker defaultTime={fromTime} is24Hour onChange={setFromTime} />
```

Comment:
Ideally, `is24Hour` would only be set for locales that use 24 hour time, but I don't see a good API to check. (And it looks like that might add some additional logic, too, if we're getting back a string.)